From: Somnath Kotur <somnath.kotur@emulex.com>
Date: Wed, 2 May 2012 03:41:01 +0000
Subject: [PATCH] be2net: Fix EEH error reset before a flash dump completes
Patch-mainline: v3.5-rc1
Git-commit: eeb7fc7bc095546b21188e8e076a59bce73f9ca6
References: bnc#755546

An EEH error can cause the FW to trigger a flash debug dump.
Resetting the card while flash dump is in progress can cause it not to recover.
Wait for it to finish before letting EEH flow to reset the card.

Signed-off-by: Sathya Perla <Sathya.Perla@emulex.com>
Signed-off-by: Somnath Kotur <somnath.kotur@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/benet/be_main.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -3548,6 +3548,11 @@ static pci_ers_result_t be_eeh_err_detec
 
 	pci_disable_device(pdev);
 
+	/* The error could cause the FW to trigger a flash debug dump.
+	 * Resetting the card while flash dump is in progress
+	 * can cause it not to recover; wait for it to finish
+	 */
+	ssleep(30);
 	return PCI_ERS_RESULT_NEED_RESET;
 }
 
